src/test_reconciler.ts

```diff
@@ -72,16 +71,16 @@ export default class TestReconciler {
     return assertions.map(assertion => {
       // Failure messages seems to always be an array of one item
       const message = assertion.failureMessages && assertion.failureMessages[0];
-      let short = null;
-      let terse = null;
-      let line = null;
+      let short: string | undefined;
+      let terse: string | undefined;
+      let line: number | undefined;
```

Comment:
The change from `null` to `undefined` stands out to me as a potentially risky API change. The risk is that a caller uses all three types from the Flow Maybe type (`type | null | void (undefined)`) to differentiate between values that aren't in the map/object, are invalid, or are a valid number. It makes me worry that I'd/we'd have to do a more thorough code review or testing.

In this particular case, the caller seems fine but there's still the rough edge of how we're using `this.lineOfError(...) || undefined` to do ... something. What would ever be falsy enough to end up as `undefined`?! I actually had to run this in the Dev Tools console just to check the falsy precedence:

<img width="168" alt="image" src="https://user-images.githubusercontent.com/2585460/73150769-f7ce0400-4095-11ea-9f33-c6a18d5b5fa0.png">

I guess I learned (or relearned) that. ☺️  Since I'll probably forget that the next time I look at it, I'd probably suggest going for:

```diff
-         line = this.lineOfError(message, filename) || undefined;
+         line = this.lineOfError(message, filename);
```
with `lineOfError` returning `undefined`. I'm not a fan of the magic numbers like `[1]` in there so I'd hope to see something like:
```diff
-   lineOfError(message: string, filePath: string): number | null {
-     const filename = path.basename(filePath);
-     const restOfTrace = message.split(filename, 2)[1];
-     return restOfTrace ? parseInt(restOfTrace.split(':')[1], 10) : null;
-   }
+   lineOfError(message: string, filepath: string): ?number {
+     const filename = path.basename(filepath);
+     const [_msgIncludingFilename, msgAfterFilename] = message.split(filename, 2);
+     if (! msgAfterFilename) return;
+     const [_colon, lineNumber] = msgAfterFilename(':');
+     return parseInt(lineNumber, 10);
+   }
```
--

The underlying question for the folks reviewing is are we OK with migrating **and** refactoring at the same time. Here's the docs for the [Flow Maybe type](https://flow.org/en/docs/types/maybe/).